torch/onnx/_backend/core.py

```diff
@@ -0,0 +1,852 @@
+                    onnx_outputs[0], baseline_outputs
+                )
+            return onnx_outputs[0]
+
+    def compile(self, graph_module: torch.fx.GraphModule, args) -> torch.fx.GraphModule:
+        # FX graph based partitioning based on ONNX supported ops.
+        if graph_module in self._partitioner_cache:
+            partitioned_prim_graph_module = self._partitioner_cache[graph_module]
+        else:
+            prim_graph_module = graph_module
+            # TODO(wschin): this is required for removing aten::_to_copy in _replace_to_copy_with_to.
+            _replace_to_copy_with_to(prim_graph_module)
+            partitioner = CapabilityBasedPartitioner(
+                prim_graph_module,
+                self._supported_ops,
+                allows_single_node_partition=True,
+            )
+            partitioned_prim_graph_module = partitioner.partition_and_fuse()
+            self._partitioner_cache[graph_module] = partitioned_prim_graph_module
+
+            # Overriding fused_module's __call__() function with ort_acclerated_call()
+            # This loop goes through all graph partitions (each of them is an ONNX-representable graph)
+            # and override their _wrappped_call function with _ort_accelerated_call.
+            # Inside _ort_accelerated_call, the partition's graph is exported into ONNX and executed by ORT.
+            for node in partitioned_prim_graph_module.graph.nodes:
+                # TODO(wschin): use a better way to identify fused submodule
+                if node.op == "call_module" and "fused_" in node.name:
+                    fused_module = getattr(partitioned_prim_graph_module, node.name)
+                    # self.ort_acclerated_call is responsible for exporting graph to ONNX,
+                    # creating ORT session, and running ORT session.
+                    fused_module._wrapped_call = self._ort_acclerated_call
+
+        return partitioned_prim_graph_module
+
+    def __call__(
+        self, graph_module: torch.fx.GraphModule, args
+    ) -> torch.fx.GraphModule:
+        return self.compile(graph_module, args)
+
+
+def make_aot_ort(dynamic: bool = True):
+    """Wrap OrtBackend as PyTorch's AOT compiler.
+
+    Example usages:
+        import torch
+        from onnxruntime.training.torchdynamo.register_backend import make_aot_ort
+        use_dynamic = True
+        local_aot_ort, _ = make_aot_ort(dynamic = use_dynamic)
+
+        @torch._dynamo.optimize(local_aot_ort, dynamic=use_dynamic)
+        def foo(x: torch.Tensor):
+            return torch.sigmoid(x)
+
+        x = torch.rand(2, 2, dtype=torch.float)
+        torch.testing.assert_close(torch.sigmoid(x), foo(x))
+    """
+    ort_backend = OrtBackend(
+        onnx_exporter_options=ExportOptions(dynamic_shapes=dynamic)
+    )
+    return (
+        aot_autograd(
+            fw_compiler=ort_backend,
+            partition_fn=min_cut_rematerialization_partition,
```

Comment:
what does this partition do and why we need it?

Reply:
Explained in comment.